lib/minitest.rb

```diff
@@ -147,16 +154,26 @@ def self.run args = []
   # sub-classes to run.
 
   def self.__run reporter, options
-    suites = Runnable.runnables.shuffle
-    parallel, serial = suites.partition { |s| s.test_order == :parallel }
-
-    # If we run the parallel tests before the serial tests, the parallel tests
-    # could run in parallel with the serial tests. This would be bad because
-    # the serial tests won't lock around Reporter#record. Run the serial tests
-    # first, so that after they complete, the parallel tests will lock when
-    # recording results.
-    serial.map { |suite| suite.run reporter, options } +
-      parallel.map { |suite| suite.run reporter, options }
+    suites = case suite_order
+    when :random then
+      Runnable.runnables.shuffle
+    else
+      Runnable.runnables.sort_by { |r| r.name }
+    end
```

Comment:
Yeah I wasn't quite certain about that. I went back and forth, but I think it is fine to just not sort it at all.

I also noticed while try to support a `:fixed` option that Ruby actually sorts method names declared at the same context level itself so you can't list them in the order they were explicitly declared, which is interesting.

I'll change this to remove the sorting option altogether.